### PR TITLE
added cleanup_folded_bound_el

### PR DESCRIPTION
**Pull request by felicio93**
added and tested the function cleanup_folded_bound_el under utils.
This function deletes all boundary elements whose nodes (all 3) are boundary nodes.

Checked pylint score, was 8.29, now is 8.31.

### CODE REVIEW

**Comment by SorooshMani-NOAA** [comment]
@felicio93 can you please add a test file as well? Just put a bunch of points and edges and call this to make sure the right elements are removed, etc.

Also I noticed pylint fails due to issues in other files. Maybe later we need to adjust pylint to avoid failing on existing code, but for now can you add ignore for the two new `E0606` issues?

```pylint
************* Module ocsmesh.hfun.collector
ocsmesh/hfun/collector.py:542:8: R1737: Use 'yield from' directly instead of yielding each element one by one (use-yield-from)
ocsmesh/hfun/collector.py:789:35: E0606: Possibly using variable 'hfun' before assignment (possibly-used-before-assignment)
ocsmesh/hfun/collector.py:34:0: W0611: Unused box imported from shapely.geometry (unused-import)
************* Module ocsmesh.geom.collector
ocsmesh/geom/collector.py:251:35: E0606: Possibly using variable 'geom' before assignment (possibly-used-before-assignment)
```

something like ` # pylint: disable=E0606 ` at the end of those two lines. Later we have to address it properly!

**Comment by felicio93** [comment]
data and test added, plus add ignore for the two new E0606 issues

**Comment by felicio93** [comment]
@SorooshMani-NOAA all changes were added (data, test, pylint ignore).

Please let me know if there is anything else.

File: tests/api/utils.py [hunk 1]
```diff
@@ -120,6 +120,14 @@ def test_cleanup_duplicate(self):
...
             self.fail(str(e))
 
+    def test_cleanup_folded_bound_el(self):
+
+        # Open mesh that has folded boundary elements
+        p = Path(__file__).parents[1] / "data" / "before_cleanup_folded_bound_el.2dm"
+        folded_bound_el_mesh = Mesh.open(p, crs=4326)
+        
+        utils.cleanup_folded_bound_el(folded_bound_el_mesh)
+
```
Comment (SorooshMani-NOAA) [inline tests/api/utils.py:132]: @felicio93 it'd be a good idea to add assertions to the test. e.g. to check the correct elements are removed, the final number of elements are expected, the return values are correct, etc.

**Comment by felicio93** [comment]
@SorooshMani-NOAA , assertion added, passed all tests.

**Pull request merged**